app/src/main/java/io/gnosis/safe/ui/safe/add/AddSafeNameViewModel.kt

```diff
@@ -32,6 +32,7 @@ class AddSafeNameViewModel
                 updateState { AddSafeNameState(ViewAction.ShowError(it)) }
             }.onSuccess {
                 updateState { AddSafeNameState(ViewAction.CloseScreen) }
+                tracker.logSafeAdd(safeRepository.getSafes().count())
```

Comment:
Generally, I would say tracking belongs in the UI but I see that here there is really no work around that. Maybe passing it as part of the `State` update? Is that something we should try to establish a pattern around?

Reply:
why should it be part of the UI? We could also track some background actions or events. Why can't we have tracking logic in view models?

Reply:
Well maybe viewModels are fine, but I would draw the line on the `data` module. Adding tracking indiscriminately will make it harder to extract functionality in the future. So at some point we would like to move something to svalinn, this could get on the way.